### DISCUSSION TITLE
Keypress - how to get the time of a key up

### DISCUSSION THREAD

**Discussion by Aodarium** [jspsych/jsPsych#686] [Q&A]
Hi, 
for one experiment, I need my participants to hold a key for an estimated amount of time (20ms < x <400ms). For that, I would like to not trigger the 'trial ending' when a key is pressed but when it is released. So far, it seems I would have to make a new plugin for that however, is the current API, with all keyboard modules, able to provide the release time of a key?
Best regards

**Comment by jodeleeuw** [comment]
There's currently no parameter to trigger this behavior in the various `keyboard-response` plugins. And unfortunately the `getKeyboardResponse()` function that helps manage keyboard events is configured only to work with key down events. We can consider adding support for key releases in the future, but your best bet for the short term is probably to modify a plugin file.
